lib/SetAPI/expression/ExpressionSetInterfaceV1.py

```diff
@@ -27,115 +48,113 @@ def save_expression_set(self, ctx, params):
-                "least one Expression object reference."
-            )
+        if not params["data"].get("items", None):
+            raise ValueError(no_items(self.set_items_type()))
 
-        refs = []
-        for item in data["items"]:
-            refs.append(item["ref"])
+        # add 'description' and item 'label' fields if not present:
+        if "description" not in params["data"]:
+            params["data"]["description"] = ""
+
+        seen_refs = set()
+        for item in params["data"]["items"]:
             if "label" not in item:
                 item["label"] = ""
-
-        ref_list = list([{"ref": r} for r in refs])
+            if item["ref"] in seen_refs:
+                raise ValueError(no_dupes())
+            seen_refs.add(item["ref"])
 
         # Get all the genome ids from our Expression references (it's the genome_id key in
         # the object metadata). Make a set out of them.
         # If there's 0 or more than 1 item in the set, then either those items are bad, or they're
         # aligned against different genomes.
+        ref_list = [{"ref": ref} for ref in seen_refs]
         info = self.workspace_client.get_object_info3(
             {"objects": ref_list, "includeMetadata": 1}
         )
-        num_genomes = len(set([item[10]["genome_id"] for item in info["infos"]]))
-        if num_genomes == 0 or num_genomes > 1:
-            raise ValueError(
-                "All Expression objects in the set must use "
-                "the same genome reference."
-            )
+        num_genomes = len({item[10]["genome_id"] for item in info["infos"]})
+        if num_genomes != 1:
+            err_msg = same_ref(self.set_items_type())
+            raise ValueError(err_msg)
 
     def get_expression_set(self, ctx, params):
-        set_type, obj_spec = self._check_get_expression_set_params(params)
+        checked_params = self._check_get_set_params(params)
 
-        include_item_info = True if params.get(INC_ITEM_INFO, 0) == 1 else False
-
-        include_set_item_ref_paths = (
-            True if params.get(INC_ITEM_REF_PATHS, 0) == 1 else False
+        obj_spec = build_ws_obj_selector(
+            checked_params["ref"], checked_params[REF_PATH_TO_SET]
         )
-
-        ref_path_to_set = params.get(REF_PATH_TO_SET, [])
+        info = self.workspace_client.get_object_info3({"objects": [obj_spec]})
+        set_type = info["infos"][0][2]
 
         if "KBaseSets" in set_type:
             # If it's a KBaseSets type, then we know the usual interface will work...
-            return self.set_interface.get_set(
-                params["ref"],
-                include_item_info,
-                ref_path_to_set,
-                include_set_item_ref_paths,
-            )
+            return self.set_interface.get_set(**checked_params)
+
+        # get_rnaseq_expression_set
+        # ...otherwise, we need to fetch it directly from the workspace and tweak it into the
```

Comment:
this hasn't changed -- I just decreased the indent level as there's no need to do an `else` after `if ...: return`